miniscope_io/stream_daq.py

```diff
@@ -215,29 +216,41 @@ def _fpga_recv(
                 "Couldnt import OpalKelly device. Check the docs for install instructions!"
             )
         # determine length
-        if read_length is None:
+        if self.config.mode == 'RAW_RECORD':
+            read_length = int(max(self.buffer_npix) * self.nbuffer_per_fm * self.config.pix_depth) * 16 # get 16 frames
+        elif read_length is None:
             read_length = int(max(self.buffer_npix) * self.config.pix_depth / 8 / 16) * 16
 
         # set up fpga devices
         BIT_FILE = self.config.bitstream
         if not BIT_FILE.exists():
             raise RuntimeError(f"Configured to use bitfile at {BIT_FILE} but no such file exists")
         # set up fpga devices
-        dev = okDev()
-        dev.uploadBit(str(BIT_FILE))
-        dev.setWire(0x00, 0b0010)
-        time.sleep(0.01)
-        dev.setWire(0x00, 0b0)
-        dev.setWire(0x00, 0b1000)
-        time.sleep(0.01)
-        dev.setWire(0x00, 0b0)
+        if self.config.mode != 'RAW_REPLAY':
+            dev = okDev()
+            dev.uploadBit(str(BIT_FILE))
+            dev.setWire(0x00, 0b0010)
+            time.sleep(0.01)
+            dev.setWire(0x00, 0b0)
+            dev.setWire(0x00, 0b1000)
+            time.sleep(0.01)
+            dev.setWire(0x00, 0b0)
         # read loop
         cur_buffer = BitArray()
         pre = Bits(self.preamble)
         if self.config.LSB:
             pre = pre[::-1]
         while True:
-            buf = dev.readData(read_length)
+            if self.config.mode == 'RAW_REPLAY':
+                with open(self.config.test_raw_data_path, 'rb') as file:
+                    buf = bytearray(file.read())
+            else:
+                buf = dev.readData(read_length)
+            if self.config.mode == 'RAW_RECORD':
+                fpga_raw_path = 'data/fpga_raw.bin' # Not sure where to define this because we don't want to overwrite.
+                with open(fpga_raw_path, 'wb') as file:
+                    file.write(buf)
+                self.terminate.value = True
```

Comment:
yes here we will want to mock the OK device instead of having an `if else` switch in the method. 

we will also want to make `record`-ing a more generalizable switch rather than a mutually exclusive mode with normal operation, so like we would have a `capture()` method with `capture(record=True)` or `capture(record='binary')` or something